sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeCreator.scala

```diff
@@ -116,9 +116,10 @@ private [sql] object GenArrayData {
        s"final ArrayData $arrayDataName = new $genericArrayClass($arrayName);",
        arrayDataName)
     } else {
+      val numBytes = elementType.defaultSize * numElements
       val unsafeArraySizeInBytes =
         UnsafeArrayData.calculateHeaderPortionInBytes(numElements) +
-        ByteArrayMethods.roundNumberOfBytesToNearestWord(elementType.defaultSize * numElements)
+        ByteArrayMethods.roundNumberOfBytesToNearestWord(numBytes).toInt
```

Comment:
Minor: why don't we inline this instead of creating a new variable?

Reply:
The size of this line is larger than 200 bytes

Reply:
We should really inline that.